app/modules/repo_manager/repo_manager.py

```
@@ -311,7 +312,19 @@ def register_repo(
+        # Ensure the repository parent directory exists
+        repo_dir = self._metadata_path(
+            repo_name=repo_name, branch=branch, commit_id=commit_id
+        ).parent
+        os.makedirs(repo_dir, exist_ok=True)
+        
+        lock_file_path = repo_dir / ".lock"
+
+        # Use portalocker for cross-platform file locking
+        # Prevent concurrent registrations for the same repo key
+        # if another process is currently registering it.
+        with portalocker.Lock(lock_file_path, timeout=10) as locked_file:
```

Ruff (0.14.8)

Local variable `locked_file` is assigned to but never used

Remove assignment to unused variable `locked_file`

(F841)